app/controllers/concerns/flashable.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+module Flashable
+  extend ActiveSupport::Concern
+
+  def message_for_notice(invalid_records)
+    msg = "Uploading #{upload_params[:csv_type]} succeeded"
+    msg + (invalid_records.present? ? ' with warnings:' : '.')
+  end
```

Comment:
ok. lets revisit that later. eventually we should require them to fix the CSV, and with the line number based messages I think this is a good path forward. Perhaps in the warning you can say something along the lines of: 'uploading a file with invalid data will no longer work in the future, please review the warnings so that file can be better prepared for uploading in the future.'